app/Repos/EntityRepo.php

```diff
@@ -196,14 +196,15 @@ public function getPageByOldSlug($pageSlug, $bookSlug)
     }
 
     /**
-     * Get all entities of a type limited by count unless count if false.
+     * Get all entities of a type with the given permission, limited by count unless count is false.
      * @param string $type
      * @param integer|bool $count
+     * @param string $permission
      * @return Collection
      */
-    public function getAll($type, $count = 20)
+    public function getAll($type, $count = 20, $permission = 'view')
```

Comment:
I've modified the existing method. There should not be any side-effects. Let me know if you would like me to create a separate `getAllByPermission` method?